src/main/java/org/gridsuite/filter/server/repositories/proxies/expertfiler/ExpertFilterRepositoryProxy.java

```diff
@@ -62,63 +65,99 @@ public static AbstractExpertRule entityToDto(ExpertRuleEntity filterEntity) {
                         .build();
             }
             case BOOLEAN -> {
-                return BooleanExpertRule.builder()
-                        .field(filterEntity.getField())
-                        .operator(filterEntity.getOperator())
-                        .value(Boolean.parseBoolean(filterEntity.getValue()))
-                        .build();
+                if (filterEntity instanceof ExpertRuleValueEntity booleanFilterEntity) {
+                    return BooleanExpertRule.builder()
+                            .field(booleanFilterEntity.getField())
+                            .operator(booleanFilterEntity.getOperator())
+                            .value(Boolean.parseBoolean(booleanFilterEntity.getValue()))
+                            .build();
+                } else {
+                    throw new PowsyblException("Invalid entity type for BOOLEAN rule");
+                }
+
             }
```

Comment:
In my opinion, we can use casting directly, this situation does normally never occur.. idem for others

Reply:
done